### ISSUE TITLE
📎 `lint/noSuspiciousSemicolonInJSX`: disallow suspicious `;` in JSX 

### ISSUE THREAD

**Issue by nstepien** [biomejs/biome#927]
### Description

I don't know if such a lint rule exists in other tools, but this is an issue I've ran into a few times before.

Basically, I'd like to have a rule to catch semicolons that may have been introduced in the JSX after a copy/paste or refactor, as these semicolons are then rendered.

Examples of invalid code:
```jsx
return (
  <div>
    <div />;
  </div>
);

return (
  <div>
    <Component>
      <div />
    </Component>;
  </div>
);
```

Examples of valid code:
```jsx
return (
  <div>
    <div />
  </div>
);

// makes it a bit more explicit that we intend to render a semicolon
return (
  <div>
    <div />
    ;
  </div>
);

// makes it a bit more explicit that we intend to render a semicolon
return (
  <div>
    <div />{';'}
  </div>
);

// use an html entity instead https://www.compart.com/en/unicode/U+003B
return (
  <div>
    <div />&#59;
  </div>
);

return (
  <div>
    {/* biome-ignore lint/??/??: rendering this semicolon is intentional */}
    <div />;
  </div>
);

return (
  <div>
    <span>;</span>
    <span />;</span />
    text; text;
    &amp;
  </div>
);

return <div />;

return <div>
  <div />
</div>;
```

It's an easy mistake to make if you don't pay attention.
![Recording 2023-11-27 at 23 16 29](https://github.com/biomejs/biome/assets/567105/5993bfce-4ea2-4119-8d46-c48d25bccc35)

**Comment by nstepien** [comment]
`noSuspiciousSemicolonInJSX` maybe?

**Comment by fujiyamaorange** [comment]
@Conaclos 
May I try this issue?

**Comment by ematipico** [comment]
All yours @fujiyamaorange
